app/scripts/views/settings/recovery_codes.js

```diff
@@ -0,0 +1,142 @@
+    this.navigate('settings/two_step_authentication');
+  },
+
+  _copyCodes() {
+    // This copies the recovery codes to clipboard by creating a tiny transparent
+    // textArea with recovery code contents. Then it executes the
+    // browser `copy` command.
+    $('<textArea class=\"recovery-code-text-area\"></textArea>').appendTo('#recovery-codes');
+    this.$('.recovery-code-text-area').html(this.recoveryCodesText);
+    this.$('.recovery-code-text-area').select();
+    this.$('.recovery-code-text-area').focus();
+    try {
+      this.window.document.execCommand('copy');
+      this._displaySuccess(t('Codes copied'));
+    } catch (err) {
+      this._displayError(t('Failed to copy codes'));
+    }
+    this.$('.recovery-code-text-area').remove();
+  },
+
+  _downloadCodes() {
+    // This dynamically creates a link with a blob data of the recovery
+    // codes and clicks it to initiate download.
+    const codeBlob = new Blob([this.recoveryCodesText], {type: 'text/plain'});
+    const filename = 'Recovery Codes.txt';
+    const href = URL.createObjectURL(codeBlob);
+    const template = `
+      <a id="recovery-code-download-link" href="${href}" download="${filename}"></a>
+    `;
+    $(template).appendTo('#recovery-codes');
+    this.window.document.getElementById('recovery-code-download-link').click();
+    this.$('#recovery-code-download-link').remove();
+  },
+
+  _printCodes() {
+    // We dynamically create a new window with recovery codes and attempt to
+    // print it.
+    const printWindow = this.window.open('', 'Print', 'height=600,width=800');
+    const renderCodes = this.recoveryCodes.map((code) => {
+      return new RecoveryCode({
+        code: code
+      }).toJSON();
+    });
+    const template = RecoveryCodePrintTemplate({
+      recoveryCodes: renderCodes
+    });
+    printWindow.document.write(template);
+    printWindow.document.close();
+    printWindow.focus();
+    printWindow.print();
+    printWindow.close();
+  },
+
+  _displaySuccess(msg) {
+    this.$('.error').addClass('hidden');
+    this.$('.modal-success').removeClass('hidden');
+    this.$('.modal-success').html(msg);
+  },
+
+  _displayError(msg) {
+    this.$('.error').removeClass('hidden');
+    this.$('.modal-success').addClass('hidden');
+    this.$('.error').html(msg);
+  },
+
+  _replaceRecoveryCodes() {
+    const account = this.getSignedInAccount();
+    return account.replaceRecoveryCodes()
+      .then((result) => {
+        this._setupRecoveryCodes(result.recoveryCodes);
+      });
+  },
+
+  _setupRecoveryCodes(codes) {
+    // Store a readable version of recovery codes so that they can
+    // be copied, printed and downloaded
+    this.recoveryCodesText = '';
+    if (codes) {
+      this.recoveryCodes = codes;
+      this.recoveryCodesText = this.recoveryCodes.join('\n');
+      this.model.set('recoveryCodes', codes);
+    }
+  },
+
+  initialize() {
+    this.on('modal-cancel', () => this._returnToTwoStepAuthentication());
```

Comment:
This seems to duplicate [this line in modal-settings-panel-mixin.js](https://github.com/mozilla/fxa-content-server/pull/5992/files#diff-b7ecfe99b7417fa4c5ec8c4cca383f13R48). Are both needed?